### PR TITLE
Add a per mutator logging option

### LINKED ISSUE #271: Infection can't report on effectiveness of mutators
| Question    | Answer
| ------------| ---------------
| Infection version | <0.9

Infection can't report on effectiveness of mutators it use. Effectiveness in a sense of number of mutations created, killed, escaped, and errorred. Such report must be mandatory for every PR with a new mutator, and can have other uses.

For example, from #243:

| Mutator          | Mutations     | Killed        | Escaped       | Errors       | 
| ---------------- | ------------- | ------------- | ------------- |------------- |
| Assign           | 437           | 281           | 27            | 1            |
| MethodCallFalse  | 1275          | 583           | 116           | 3            |
| MethodCallNull   | 1275          | 582           | 117           | 3            |
| MethodCallTrue   | 1275          | 563           | 135           | 4            |

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Adds new feature, the per mutator logging option
- [x] Covered by tests
- [x] Doc PR: https://github.com/infection/site/pull/40

Fixes #271 

Both are on php 7.0

<details>
 <summary>xdebug output</summary>
 
# Effects per Mutator

| Mutator | Mutations | Killed | Escaped | Errors | MSI | Covered MSI |
| ------- | --------- | ------ | ------- |------- | --- | ----------- |
| ArrayItem | 19 | 12 | 7 | 0 | 63| 63|
| Assignment | 1 | 0 | 0 | 0 | 0| 0|
| AssignmentEqual | 4 | 3 | 1 | 0 | 75| 75|
| BitwiseAnd | 4 | 4 | 0 | 0 | 100| 100|
| BitwiseNot | 2 | 2 | 0 | 0 | 100| 100|
| BitwiseOr | 2 | 2 | 0 | 0 | 100| 100|
| Break_ | 2 | 0 | 2 | 0 | 0| 0|
| CastArray | 7 | 5 | 2 | 0 | 71| 71|
| CastBool | 6 | 5 | 0 | 0 | 83| 100|
| CastFloat | 7 | 2 | 3 | 0 | 29| 40|
| CastInt | 11 | 7 | 2 | 0 | 64| 78|
| CastString | 3 | 2 | 1 | 0 | 67| 67|
| Continue_ | 13 | 6 | 6 | 0 | 46| 50|
| DecrementInteger | 73 | 35 | 23 | 0 | 48| 60|
| Division | 3 | 3 | 0 | 0 | 100| 100|
| Equal | 33 | 26 | 3 | 0 | 79| 90|
| FalseValue | 69 | 54 | 9 | 0 | 80| 86|
| Foreach_ | 59 | 41 | 13 | 0 | 71| 76|
| FunctionCall | 25 | 19 | 4 | 0 | 76| 83|
| GreaterThan | 8 | 3 | 5 | 0 | 38| 38|
| GreaterThanNegotiation | 8 | 6 | 2 | 0 | 75| 75|
| GreaterThanOrEqualTo | 5 | 2 | 2 | 0 | 40| 50|
| GreaterThanOrEqualToNegotiation | 5 | 2 | 2 | 0 | 40| 50|
| Identical | 32 | 8 | 20 | 0 | 25| 29|
| IdenticalEqual | 32 | 4 | 24 | 0 | 13| 14|
| Increment | 7 | 6 | 1 | 0 | 86| 86|
| IncrementInteger | 40 | 16 | 12 | 0 | 40| 57|
| IntegerNegation | 8 | 1 | 1 | 0 | 13| 50|
| LessThan | 6 | 0 | 5 | 0 | 0| 0|
| LessThanNegotiation | 6 | 0 | 4 | 0 | 17| 20|
| LessThanOrEqualTo | 3 | 2 | 1 | 0 | 67| 67|
| LessThanOrEqualToNegotiation | 3 | 3 | 0 | 0 | 100| 100|
| LogicalAnd | 39 | 23 | 14 | 0 | 59| 62|
| LogicalNot | 117 | 100 | 12 | 0 | 85| 89|
| LogicalOr | 33 | 21 | 10 | 0 | 64| 68|
| Minus | 6 | 5 | 1 | 0 | 83| 83|
| Modulus | 1 | 0 | 1 | 0 | 0| 0|
| MulEqual | 1 | 0 | 0 | 0 | 0| 0|
| Multiplication | 5 | 5 | 0 | 0 | 100| 100|
| NewObject | 64 | 62 | 1 | 0 | 97| 98|
| NotEqual | 4 | 3 | 0 | 0 | 75| 100|
| NotIdentical | 17 | 7 | 7 | 0 | 41| 50|
| NotIdenticalNotEqual | 17 | 5 | 9 | 0 | 29| 36|
| OneZeroFloat | 6 | 6 | 0 | 0 | 100| 100|
| OneZeroInteger | 89 | 45 | 18 | 0 | 52| 72|
| Plus | 8 | 5 | 1 | 0 | 63| 83|
| PregQuote | 1 | 1 | 0 | 0 | 100| 100|
| ProtectedVisibility | 17 | 4 | 1 | 0 | 24| 80|
| PublicVisibility | 146 | 112 | 22 | 0 | 77| 84|
| Spaceship | 1 | 1 | 0 | 0 | 100| 100|
| Throw_ | 34 | 19 | 2 | 0 | 56| 90|
| TrueValue | 51 | 29 | 15 | 0 | 57| 66|
</details>

<details>
 <summary>phpdbg output</summary>
 
# Effects per Mutator

| Mutator | Mutations | Killed | Escaped | Errors | MSI | Covered MSI |
| ------- | --------- | ------ | ------- |------- | --- | ----------- |
| ArrayItem | 19 | 14 | 5 | 0 | 74| 74|
| Assignment | 1 | 0 | 0 | 0 | 0| 0|
| AssignmentEqual | 4 | 3 | 1 | 0 | 75| 75|
| BitwiseAnd | 4 | 4 | 0 | 0 | 100| 100|
| BitwiseNot | 2 | 2 | 0 | 0 | 100| 100|
| BitwiseOr | 2 | 2 | 0 | 0 | 100| 100|
| Break_ | 2 | 2 | 0 | 0 | 100| 100|
| CastArray | 7 | 6 | 1 | 0 | 86| 86|
| CastBool | 6 | 5 | 0 | 0 | 83| 100|
| CastFloat | 7 | 5 | 0 | 0 | 71| 100|
| CastInt | 11 | 8 | 1 | 0 | 73| 89|
| CastString | 3 | 3 | 0 | 0 | 100| 100|
| Continue_ | 13 | 6 | 6 | 0 | 46| 50|
| DecrementInteger | 73 | 54 | 4 | 0 | 74| 93|
| Division | 3 | 3 | 0 | 0 | 100| 100|
| Equal | 33 | 28 | 1 | 0 | 85| 97|
| FalseValue | 69 | 62 | 1 | 0 | 91| 98|
| Foreach_ | 59 | 52 | 2 | 0 | 90| 96|
| FunctionCall | 25 | 22 | 1 | 0 | 88| 96|
| GreaterThan | 8 | 5 | 3 | 0 | 63| 63|
| GreaterThanNegotiation | 8 | 7 | 1 | 0 | 88| 88|
| GreaterThanOrEqualTo | 5 | 4 | 0 | 0 | 80| 100|
| GreaterThanOrEqualToNegotiation | 5 | 4 | 0 | 0 | 80| 100|
| Identical | 32 | 28 | 0 | 0 | 88| 100|
| IdenticalEqual | 32 | 26 | 2 | 0 | 81| 93|
| Increment | 7 | 7 | 0 | 0 | 100| 100|
| IncrementInteger | 40 | 27 | 1 | 0 | 68| 96|
| IntegerNegation | 8 | 2 | 0 | 0 | 25| 100|
| LessThan | 6 | 5 | 0 | 0 | 83| 100|
| LessThanNegotiation | 6 | 4 | 0 | 0 | 83| 100|
| LessThanOrEqualTo | 3 | 3 | 0 | 0 | 100| 100|
| LessThanOrEqualToNegotiation | 3 | 3 | 0 | 0 | 100| 100|
| LogicalAnd | 39 | 35 | 2 | 0 | 90| 95|
| LogicalNot | 117 | 110 | 2 | 0 | 94| 98|
| LogicalOr | 33 | 28 | 3 | 0 | 85| 90|
| Minus | 6 | 6 | 0 | 0 | 100| 100|
| Modulus | 1 | 1 | 0 | 0 | 100| 100|
| MulEqual | 1 | 0 | 0 | 0 | 0| 0|
| Multiplication | 5 | 5 | 0 | 0 | 100| 100|
| NewObject | 64 | 63 | 0 | 0 | 98| 100|
| NotEqual | 4 | 3 | 0 | 0 | 75| 100|
| NotIdentical | 17 | 13 | 1 | 0 | 76| 93|
| NotIdenticalNotEqual | 17 | 11 | 3 | 0 | 65| 79|
| OneZeroFloat | 6 | 6 | 0 | 0 | 100| 100|
| OneZeroInteger | 89 | 59 | 4 | 0 | 67| 94|
| Plus | 8 | 6 | 0 | 0 | 75| 100|
| PregQuote | 1 | 1 | 0 | 0 | 100| 100|
| ProtectedVisibility | 17 | 5 | 0 | 0 | 29| 100|
| PublicVisibility | 146 | 134 | 0 | 0 | 92| 100|
| Spaceship | 1 | 1 | 0 | 0 | 100| 100|
| Throw_ | 34 | 21 | 0 | 0 | 62| 100|
| TrueValue | 51 | 39 | 5 | 0 | 76| 89|
</details>

### CODE REVIEW

File: src/Logger/PerMutatorLogger.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    protected function getLogLines(): array
+    {
+        $this->setUpPerCalculatorMutator();
+
+        $logs = [];
+
+        $logs[] = "# Effects per Mutator\n";
+
+        $logs[] = '| Mutator | Mutations | Killed | Escaped | Errors | Timed Out | MSI | Covered MSI |';
```
Comment (sidz) [inline src/Logger/PerMutatorLogger.php]: \o 
couldcan we use https://symfony.com/doc/current/components/console/helpers/table.html helper here?

Comment (borNfreee) [reply src/Logger/PerMutatorLogger.php]: @sidz but current PR is for markdown, not for console

Comment (BackEndTea) [reply src/Logger/PerMutatorLogger.php]: A console version of this may be a nice idea for another PR, it could be usefull to have that info printed on CI.

**Comment by BackEndTea** [comment]
Changed per-mutator to perMutator

**Comment by borNfreee** [comment]
Thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @borNfreee @sidz

**Pull request merged**